homeassistant/components/chacon_dio/switch.py

```diff
@@ -0,0 +1,74 @@
+"""Switch Platform for Chacon Dio REV-LIGHT and switch plug devices."""
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api.const import DeviceTypeEnum
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ChaconDioConfigEntry
+from .entity import ChaconDioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ChaconDioConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Chacon Dio switch devices."""
+    data = config_entry.runtime_data
+    client = data.client
+
+    async_add_entities(
+        ChaconDioSwitch(client, device)
+        for device in data.list_devices
+        if device["type"]
+        in (DeviceTypeEnum.SWITCH_LIGHT.value, DeviceTypeEnum.SWITCH_PLUG.value)
```

Comment:
If it's a light, shouldnt it also be a light platform?

Reply:
It is a wall switch whose usage is for simple light : https://chacon.com/en/switches-connected/1965-connected-plug-dio-connect-5411478549505.html
I am perhaps wrong thinking the correct platform is a switch for this type of device ?

Reply:
Oh then it is a switch